src/app/modals/modals.service.ts

```diff
@@ -80,9 +81,14 @@ export class ModalsService {
         if (!this.wasAlreadyOpen(modal)) {
           this.log.d(`next modal: ${modal}`);
           this.modal = this.messages[modal];
-          this.message.next({modal: this.modal, data: data});
+          dialogRef.componentInstance.open(this.modal, {data: data});
+          // this.message.next({modal: this.modal, data: data});
           this.isOpen = true;
-          this.enableClose = true;
+          dialogRef.componentInstance.enableClose = this.enableClose = true;
+          dialogRef.disableClose = this.enableClose;
```

Comment:
does this need to observe the state as well?

Reply:
dialogRef.componentInstance.enableClose = true. is req. to set value on first time other will be remove.